lib/Pusher.php

```diff
@@ -533,7 +533,7 @@ public function socket_auth($channel, $socket_id, $custom_data = false)
         $this->validate_channel($channel);
         $this->validate_socket_id($socket_id);
 
-        if ($custom_data === true) {
```

Comment:
I think `if ($custom_data)` is enough since a non-empty string would be a truthy value. If it's not a string but still truthy then it should fail during the concatenation.